common/tools/dev-tool/src/util/printCommandUsage.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license
+
+import { CommandInfo, CommandLoader } from "./commandModule";
+
+const hasKey = (o: any, k: string) => Object.prototype.hasOwnProperty.call(o, k);
+
+/**
+ * The stack of subcommands executed so far
+ *
+ * Used in help output.
+ */
+export const commandStack: string[] = [];
+
+/**
+ * Prints the help information for a given command
+ *
+ * @param info CommandInfo structure to use in printing command information
+ * @param subCommands optional CommandLoader structure that defines subcommand usage
+ * @param println optionally override the default printer
+ */
+export async function printCommandUsage(
+  info: CommandInfo,
+  subCommands?: CommandLoader,
+  println: (...values: string[]) => void = console.log
+) {
+  println(`${info.name} - ${info.description}\n`);
+  println(
+    `Usage: ${commandStack.join(" ")} [OPTIONS] ${
+      subCommands !== undefined ? "COMMAND [... COMMAND-OPTIONS]" : ""
+    }\n`
+  );
+
+  // OPTIONS info
+  println("Options:");
+  println("  --help\t<boolean> display this help message");
+  if (info.options) {
+    for (const k in info.options) {
```

Comment:
I thought for/in was generally discouraged and you can iterate over keys using the result of Object.keys (or Object.values/Object.entries as appropriate)

Reply:
old habit to use `for in` and guard with `hasOwnProperty`... I've changed all of these to use Object.keys